File: front-end/src/app/shared/models/match-model.ts
import StatsModel from './stasts-model';
import ParticipantModel from './participant-model';
import { ChampionModel } from './champion-model';
import TeamModel from './team-model';

export default class MatchModel {
    champion: number;
    championModel: ChampionModel;
    gameId: number;
    lane: string;
    platformId: string;
    queue: number;
    role: string;
    season: number;
    timestamp: number;
    stats: StatsModel;
    participants: ParticipantModel[];
    gameCreation: number;
    gameDuration: number;
    gameMode: string;
    gameType: string;
    gameVersion: string;
    mapId: number;
    participantIdentities: any;
    queueId: number;
    seasonId: number;
    teams: TeamModel[];
}

File: back-end/LeagueAppAPI/LeagueAppAPI/Models/MatchModel.cs

namespace LeagueAppAPI.Models
{
    public class MatchModel
    {
        public long gameId { get; set; }
        public int champion { get; set; }
        // championModel: ChampionModel;
        public string lane { get; set; }
        public string platformId { get; set; }
        public int queue { get; set; }
        public string role { get; set; }
        public int season { get; set; }
        public long timestamp { get; set; }
        // stats: StatsModel;
        // participants: ParticipantModel[];
        public int gameCreation { get; set; }
        public int gameDuration { get; set; }
        public string gameMode { get; set; }
        public string gameType { get; set; }
        public string gameVersion { get; set; }
        public int mapId { get; set; }
        // participantIdentities: any;
        public int queueId { get; set; }
        public int seasonId { get; set; }
        // teams: TeamModel[];
    }
}
